core/app/models/spree/stock/allocator/base.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Spree
+  module Stock
+    module Allocator
+      class Base
+        attr_reader :availability
+
+        def initialize(availability)
+          @availability = availability
+        end
+
+        def allocate_inventory(desired)
+          # Allocate any available on hand inventory
+          on_hand, desired = allocate_on_hand(desired)
```

Comment:
Does `allocate_on_hand` need to return the new desired inventory? I think we can easily compute it ourselves:
```ruby
on_hand = allocate_on_hand(desired)
desired -= on_hand
```
This relieves `allocate_on_hand` of one responsibility and makes it easier to write custom allocators.

What do you think? Is this doable?